src/ui/Base/QueryBuilder.ts

```diff
@@ -282,8 +282,21 @@ export class QueryBuilder {
 
   /**
    * Specifies an array of request for the MLFacet component.
+   * Can not be used at the same time as [`groupByRequests`]{@link QueryBuilder.groupByRequests}
+   * Use the [`addFacetRequest`]{@link QueryBuilder.addFacetRequest} method to define this value
    */
-  public facets: IFacetRequest[] = [];
+  public facetRequests: IFacetRequest[] = undefined;
+
+  /**
+   * Defines and adds a value to the [`facetRequests`]{@link QueryBuilder.facetRequests}
+   */
+  public addFacetRequest(facetRequest: IFacetRequest) {
+    if (!this.facetRequests) {
```

Comment:
Should there be a check here on whether `groupByRequests.length !== 0`? And maybe logging a warning when the condidtion is true saying that the facetRequest will not be added?

At the moment, it is possible to add groupByRequests and facetRequests, with the decision happening when validating on 523. The tricky thing with that condition is it has a hidden preference for groupByRequests that can only be discovered by opening the file and reading the code.

i.e. if we have facetRequests and groupByRequests, `validateGroupBy` will return the groupByRequests.

Also, in the above case, won't `this.facetRequests` return the facetRequests, so both properties are truthy which is invalid?

Now that I think about all this, why not move these restrictions to the API? If both are arrays are populated, use groupBy. Otherwise use the property with the populated array.